its/it/azdo.ts

```diff
@@ -0,0 +1,41 @@
+import * as vm from "azure-devops-node-api";
+import { Build, BuildResult, BuildStatus } from "azure-devops-node-api/interfaces/BuildInterfaces";
+import { AZDO_BASE_URL, AZDO_ORGANIZATION, AZDO_PROJECT } from "../constant";
+import { getBranch, loadEnvironmentVariables } from "./env";
+
+export function getAzdoApi(): vm.WebApi {
+  const env = loadEnvironmentVariables();
+  const credentialHandler = vm.getPersonalAccessTokenHandler(env.AZURE_TOKEN);
+  return new vm.WebApi(AZDO_BASE_URL + AZDO_ORGANIZATION, credentialHandler);
+}
+
+export async function runPipeline(azdoApi: vm.WebApi, pipelineName: string): Promise<Build> {
+  const azdoBuildApi = await azdoApi.getBuildApi();
+  const definitions = await azdoBuildApi.getDefinitions(AZDO_PROJECT, pipelineName);
+  if (definitions.length === 0) {
+    throw new Error(`No pipeline found with name ${pipelineName}`);
+  }
+  const build = await azdoBuildApi.queueBuild(
+    {
+      definition: { id: definitions[0].id as number },
+      project: definitions[0].project,
+      sourceBranch: getBranch(),
+    },
+    AZDO_PROJECT,
+  );
+  const buildId = build.id as number;
+  const projectId = build.project?.id as string;
+
+  console.log(`Build ${buildId} queued`);
```

Comment:
the output looks like

```
Running pipeline pipeline-sonarcloud-v3-dotnet-embedded-windows
Build 103467 queued
```

What should I do with the `buildId` ? is it accessible?

I see `pipeline-sonarcloud-v3-dotnet-embedded-windows` matches the name of  
https://dev.azure.com/sonarsource/SonarScannerAzdo/_build?definitionId=167

But I only see manually triggered pipelines there 🤔 . Whats the relationship?

Reply:
These pipelines are shown as "manually triggered" because they are triggered by another pipeline ([this one](https://dev.azure.com/sonarsource/SonarScannerAzdo/_build?definitionId=160)) - We are making an API call to AzDO to ask to run this pipeline against the correct branch

Reply:
The build id is what's in the URL eg https://dev.azure.com/sonarsource/SonarScannerAzdo/_build/results?buildId=103473&view=results

Reply:
Makes sense, one thing I am missing is for all the other configurations that we aren't running through the pipeline automatically, e.g. `pipeline-sonarcloud-v1-cli-6.2.1.4610-unix` or any non-latest task; 

These are not exposed to azdo anywhere atm right? Is there a way atm to trigger these with low effort?

Reply:
I think there is, I can maybe create a pull request for it? In addition we can use this specific pull request/commit as a documentation/example of how to add a test case?